clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -179,6 +187,10 @@ public void stop() {
      * @throws Exception If there was an error while sending.
      */
     public void send(String method, Object... args) throws Exception {
+        if (connectionState != HubConnectionState.CONNECTED) {
+            throw new Exception("Cannot send data if the connection is not in the 'Connected' State.");
```

Comment:
This will be `HubException` once https://github.com/aspnet/SignalR/pull/2821 is merged.

Reply:
Should the message be `Connected` or `CONNECTED` since that's the enum name

Reply:
Okay, we should update the TS client for consistency

Reply:
https://github.com/aspnet/SignalR/blob/8c9093332b90e8a64497ebb8bc06ed84a99b1615/clients/ts/signalr/src/HttpConnection.ts#L95-L98

Reply:
💯